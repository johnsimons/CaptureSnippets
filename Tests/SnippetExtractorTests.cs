using CaptureSnippets;
using NUnit.Framework;
using ObjectApproval;

[TestFixture]
public class SnippetExtractorTests
{
    [Test]
    public void CanExtractFromXml()
    {
        var input = @"
  <!-- startcode CodeKey -->
  <configSections/>
  <!-- endcode -->";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Test]
    public void UnClosedSnippet()
    {
        var input = @"
  <!-- startcode CodeKey -->
  <configSections/>";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }
    [Test]
    public void UnClosedRegion()
    {
        var input = @"
  #region CodeKey
  <configSections/>";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Test]
    public void CanExtractFromRegion()
    {
        var input = @"
  #region CodeKey 4
  The Code
  #endregion";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }



    [Test]
    public void CanExtractWithNoTrailingCharacters()
    {
        var input = @"
  // startcode CodeKey
  the code
  // endcode ";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Test]
    public void CanExtractWithMissingSpaces()
    {
        var input = @"
  <!--startcode CodeKey-->
  <configSections/>
  <!--endcode-->";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }
    [Test]
    public void CanExtractWithTrailingWhitespace()
    {
        var input = @"
  // startcode CodeKey
  the code
  // endcode   ";
        var snippets = SnippetExtractor.FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }
}