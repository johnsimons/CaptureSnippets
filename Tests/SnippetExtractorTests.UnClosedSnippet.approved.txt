[
    {
        "StartRow": 2,
        "EndRow": 0,
        "Value": "SNIPPET WAS NOT CLOSED",
        "Key": "codekey",
        "Version": null,
        "Language": "",
        "File": null,
        "IsUnclosed": true
    }
]
